### PR TITLE
Refactor/workspace epic

### PR DESCRIPTION
**Pull request by my-code-is-lit**

### CODE REVIEW

File: src/client/src/apps/MainRoute/store/combineReducers.ts [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
 import { reducers } from 'apps/MainRoute/data'
 import { analyticsReducer } from '../widgets/analytics/index'
 import { blotterReducer } from '../widgets/blotter/index'
 import { spotTileDataReducer } from '../widgets/spotTile/index'
 
+const customReducers = platform.customReducers
+
 const rootReducer = combineReducers({
   ...reducers,
+  ...customReducers,
```
Comment (limerentfox) [inline src/client/src/apps/MainRoute/store/combineReducers.ts]: not sure if I'm onboard with prefixing things with custom when the rest are semantically named

File: src/client/src/apps/MainRoute/store/configureStore.ts [hunk 2]
```diff
@@ -16,8 +17,11 @@ import { createSpotTileEpic } from '../widgets/spotTile/index'
 
 import rootReducer from './combineReducers'
 
+const customEpics = platform.customEpics
+
 export default function configureStore(dependencies: ApplicationDependencies) {
   const epics = [
+    ...customEpics,
```
Comment (limerentfox) [inline src/client/src/apps/MainRoute/store/configureStore.ts]: ditto

File: src/client/src/rt-platforms/openFin/adapter/openFin.ts [hunk 3]
```diff
@@ -207,23 +220,19 @@ async function appRestoreHandler(workspaceApp: workspaces.WorkspaceApp, store: S
...
       )
 
       // we need to 'remove' the child window from the main window
-      store.dispatch(
-        LayoutActions.updateContainerVisibilityAction({
-          name: win.name,
-          display: false,
-        }),
-      )
+      callback({
```
Comment (limerentfox) [inline src/client/src/rt-platforms/openFin/adapter/openFin.ts]: I don't know why you took this out of the store. Are you relying solely on openfin to save these layouts? if anything put it in localstorage. And I thought workspaces was just openfin. Why extract a platform specfic feature and generalize the naming?

Comment (my-code-is-lit) [reply src/client/src/rt-platforms/openFin/adapter/openFin.ts]: It wasn't taken out of the store, Bhavesh didn't want the entire store object passed to OpenFin adapter

Comment (limerentfox) [reply src/client/src/rt-platforms/openFin/adapter/openFin.ts]: got it, alrighty then

**Pull request merged**